scripts/bump_version.py

```diff
@@ -13,7 +13,8 @@
 version_part = parser.parse_args().version_part
 
 new_major, new_minor, new_patch = (
-    int(current) + (part == version_part)
+    (int(current) + (part == version_part))
+    if part <= version_part else 0
```

Comment:
I'm really really struggling to figure this code out about how it works. I've pulled it down locally and put in a bunch of print statements and I'm a bit closer but still not quite there. I'm finding it very complex by being done mostly a one liner. I would suggest a potential rewrite that really focusses on readability. I came with some rough pseudocode as an example


```
if bump == 'patch':
    new_patch = current_patch + 1
    new_minor = current_minor
    new_major = current_major

if bump == 'minor':
    new_patch = 0
    new_minor = current_minor + 1
    new_major = current_major

if bump == 'major':
    new_patch = 0
    new_minor = 0 
    new_major = current_major + 1
```

Thoughts? This isn't necessarily going to be a blocker if you feel strongly that the current approach is OK in terms of complexity

Reply:
The secret is that `'major'`, `'minor'` and `'patch'` happen to stay order of precedence when you sort them alphabetically 😉 

How about something like this as somewhere in between the two?

```python
new_major, new_minor, new_patch = {
    'major': (current_major + 1, 0, 0),
    'minor': (current_major, current_minor + 1, 0),
    'patch': (current_major, current_minor, current_patch + 1),
}[bump]
```

Reply:
Love it!

Reply:
🥳  I’ve changed it to that and squashed